src/main/scala/com/campudus/tableaux/cache/CacheClient.scala

```diff
@@ -0,0 +1,99 @@
+package com.campudus.tableaux.cache
+
+import com.campudus.tableaux.database.domain.DomainObject
+import com.campudus.tableaux.database.model.TableauxModel.{ColumnId, RowId, TableId}
+import io.vertx.core.eventbus._
+import io.vertx.core.{AsyncResult, Handler, Vertx}
+import io.vertx.scala.VertxExecutionContext
+import org.vertx.scala.core.json._
+
+import scala.concurrent.Future
+
+object CacheClient {
+  def apply(vertx: Vertx): CacheClient = {
+    new CacheClient(vertx: Vertx)
+  }
+}
+
+class CacheClient(override val _vertx: Vertx) extends VertxExecutionContext {
+
+  import io.vertx.scala.FunctionConverters._
+
+  type AsyncMessage[A] = Handler[AsyncResult[Message[A]]]
+
+  private val eventBus = _vertx.eventBus()
+
+  private def key(tableId: TableId, columnId: ColumnId, rowId: RowId) = {
+    Json.obj(
+      "tableId" -> tableId,
+      "columnId" -> columnId,
+      "rowId" -> rowId
+    )
+  }
+
+  def setCellValue(tableId: TableId, columnId: ColumnId, rowId: RowId, value: Any): Future[Unit] = {
+    val encodedValue = value match {
+      case v: Seq[_] => v.map({
+        case e: DomainObject => e.getJson
+        case e => e
+      })
+      case v: DomainObject => v.getJson
+      case _ => value
+    }
+
+    val obj = key(tableId, columnId, rowId).copy().mergeIn(Json.obj("value" -> encodedValue))
+
+    val options = new DeliveryOptions()
+      .setSendTimeout(200)
+
+    eventBus.send(CacheVerticle.ADDRESS_SET, obj, options)
+
+    //(eventBus.send(CacheVerticle.ADDRESS_SET, obj, options, _: AsyncMessage[JsonObject]))
+    //  .map(_ => ())
+
+    Future.successful(())
```

Comment:
If it's fire-and-forget, I wouldn't return anything other than `Unit`. This would make the "this is a side-effecting function" more obvious. And you would not need the "else" part in the later part (comment below, search for "Kartoffelpürre")